From 614a2c844461fcb74893375b82e1e25b3bd0c25a Mon Sep 17 00:00:00 2001
From: Jiri Pirko <jpirko@redhat.com>
Date: Tue, 8 Jul 2014 06:15:37 -0400
Subject: [netdrv] bonding: remove no longer relevant vlan warnings

Message-id: <1404800137-3945-2-git-send-email-jpirko@redhat.com>
Patchwork-id: 85294
O-Subject: [patch rhel7 net 2/2] bonding: remove no longer relevant vlan warnings
Bugzilla: 1048814
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Nikolay Aleksandrov <nikolay@redhat.com>
RH-Acked-by: Veaceslav Falico <vfalico@redhat.com>

BZ1048814
https://brewweb.devel.redhat.com/taskinfo?taskID=7658069
Tested on my testing machine

commit (net-next) e721f87d806f2a959d6a530be18dabee6097aa79
Author: Jiri Pirko <jiri@resnulli.us>
Date:   Wed Jul 2 11:50:36 2014 +0200

    bonding: remove no longer relevant vlan warnings

    These warnings are no longer relevant. Even when last slave is
    removed, there is a valid address assigned to bond (random).
    The correct functionality of vlans is ensured by maintaining unicast
    list in vlan_sync_address().

Signed-off-by: Jiri Pirko <jpirko@redhat.com>
Signed-off-by: Jiri Benc <jbenc@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/bonding/bond_main.c b/drivers/net/bonding/bond_main.c
index b1432d9..b0507d4 100644
--- a/drivers/net/bonding/bond_main.c
+++ b/drivers/net/bonding/bond_main.c
@@ -2056,13 +2056,6 @@ static int __bond_release_one(struct net_device *bond_dev,
  if (bond->slave_cnt == 0) {
   bond_set_carrier(bond);
   eth_hw_addr_random(bond_dev);
-
-  if (bond_vlan_used(bond)) {
-   pr_warning("%s: Warning: clearing HW address of %s while it still has VLANs.\n",
-       bond_dev->name, bond_dev->name);
-   pr_warning("%s: When re-adding slaves, make sure the bond's HW address matches its VLANs'.\n",
-       bond_dev->name);
-  }
  }
 
  write_unlock_bh(&bond->lock);
-- 
1.7.1